contracts/SpokePool.sol

```diff
@@ -151,10 +157,35 @@ abstract contract SpokePool is SpokePoolInterface, Testable, Lockable, MultiCall
         _;
     }
 
+    modifier unpausedDeposits() {
+        require(!pausedDeposits, "Paused deposits");
+        _;
+    }
+
+    modifier unpausedFills() {
+        require(!pausedFills, "Paused fills");
+        _;
+    }
+
     /**************************************
      *          ADMIN FUNCTIONS           *
      **************************************/
 
+    /**
+     * @notice Pauses deposit and fill functions. This is intended to be used during upgrades or when
+     * something goes awry.
+     * @param pause true if the call is meant to pause the system, false if the call is meant to unpause it.
+     */
+    function pauseDeposits(bool pause) public override onlyAdmin nonReentrant {
+        pausedDeposits = pause;
+        emit PausedDeposits(pausedDeposits);
+    }
+
+    function pauseFills(bool pause) public override onlyAdmin nonReentrant {
+        pausedFills = pause;
+        emit PausedFills(pausedDeposits);
```

Comment:
The event here looks incorrect - should emit the value of pausedFills instead of pausedDeposits.

Reply:
great catch